src/scraper/scraper.py

```diff
@@ -0,0 +1,63 @@
+"""
+    Main module interface
+"""
+from scrapy.crawler import CrawlerProcess
+from scrapy         import signals
+from typing         import List
+from .settings      import URL_TO_SPIDERS, CRAWLER_SETTINGS
+
+_process = CrawlerProcess(CRAWLER_SETTINGS)
+
+def scrape(url : str) -> List[dict]:
```

Comment:
Could you create unit tests for the scrape function?

Reply:
I want to, I just don't know how yet, since I need to find a way to test a web scraper and decoupling tests from the http request to the page being scraped. I have a question, where should I write the test once they've been wrote? With nox tests? or in its own file/folder within the scraper module?

Reply:
We have been adding our tests (not necessarily unit tests) under `tests/data` folder since our existing modules are in `src/data`.  Like mirroring the `src` folder structure.

Reply:
Put the tests in their own folder, for now, since the scraper is in `/src/bla-bla-scraper`, put the tests in a `tests` folder at the top level (e.g. `/tests/bla-bla-specs`). This way, our tests *use* our library—instead of *being part* of the library.

My two cents regarding testing. You do not test the scraper (Scrapy already does that), and you do not test that the webpage works—because websites are fragile and monitoring that scrapers work and alerting somebody is not part of the scope of this phase.

You should test the expected behavior of all the code that you wrote. By thinking hard about testing the implemented code, you decouple external modules from the core ones. For example, by decoupling the HTTP adapter and the Scraping Engine from our scraping code itself. Remember that great testability is a symptom of a great design.

That way, we can mock parts of the Scrapy framework, HTTP responses, or any outsider code. Only if it's needed that is :)

However, this is hard to do! Try not to get frustrated if you are struggling. It's normal.

Reply:
+1 for the tests, I would also suggest that instead of writing the tests in `src/scraper/tests`, you write them in `/tests`